src/main/java/org/jenkinsci/plugins/workflow/cps/nodes/StepNode.java

```diff
@@ -1,22 +1,18 @@
 package org.jenkinsci.plugins.workflow.cps.nodes;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import org.jenkinsci.plugins.workflow.graph.FlowNode;
 import org.jenkinsci.plugins.workflow.steps.Step;
 import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
 
 import javax.annotation.CheckForNull;
 
 /**
- * Optional interface for {@link FlowNode} that has associated {@link StepDescriptor}
- *
- * @author Kohsuke Kawaguchi
+ * Optional interface for {@link FlowNode} that has associated {@link StepDescriptor}.
+ * Retained for binary and API compatibility but the interface has been pushed up a level into the
+ *  workflow-api plugin to allow broader access.
  */
-public interface StepNode {
-    /**
-     * Returns the descriptor for {@link Step} that produced this flow node.
-     *
-     * @return
-     *      null for example if the descriptor that created the node has since been uninstalled.
-     */
-    @CheckForNull StepDescriptor getDescriptor();
+@Deprecated
```

Comment:
:NIT: would be nice to java javadoc suggesting to use the workflow-api interface instead.